nova/virt/vmwareapi/vm_util.py

```diff
@@ -1434,47 +1436,47 @@ def get_vm_state(session, instance):
     return constants.POWER_STATES[vm_state]
 
 
-def _get_host_reservations(host_reservations_map, host_moref, host_vcpus,
-                           host_memory_mb):
+def _set_host_reservations(stats, host_reservations_map, host_moref):
     """Compute the number of vcpus and memory in MB reserved for the host from
     the configured reservations or the default.
 
     For every host, both vcpus and memory can be given as static number or in
     percent, with static number taking priority.
     """
-    reservations = {
-        'vcpus': 0,
-        'memory_mb': 0
-    }
+    host_vcpus = stats["vcpus"]
+    host_memory_mb = stats["memory_mb"]
+    stats["vcpus_reserved"] = 0
+    stats["memory_mb_reserved"] = 0
 
     default_key = _HOST_RESERVATIONS_DEFAULT_KEY
     host_reservations = host_reservations_map.get(default_key, {})
     group_reservations = host_reservations_map.get(host_moref.value, {})
     for key in ['vcpus', 'vcpus_percent', 'memory_mb', 'memory_percent']:
         if key in group_reservations:
             host_reservations[key] = group_reservations[key]
+
     if not host_reservations:
-        return reservations
+        return stats
```

Comment:
Why does it return something? I would think it's cleaner to not return anything, if we are supposed to edit the given object. Otherwise, one might think this is a copy or something.

Reply:
I'll remove that